services/queue/README.md

```diff
@@ -21,3 +21,91 @@ For S3 we have a dummy bucket called `test-bucket-for-any-garbage` which stores
 Mozilla developers can get access from a taskcluster developer, or you can setup a custom a bucket and overwrite the bucket name as well as the credentials.
 
 Same thing applies for azure, though it's not as nicely scoped, and doesn't clean up on its own.
+
+## Queue internals
+
+Queue service is using internal queues to manage task lifecycle. Those are:
+
+- pending queue - scheduled tasks ready to be claimed
+- claim queue - tasks that were claimed, and are awaiting to be resolved or failed with `claim-expired`
+- deadline queue - task deadlines, to help resolve task graph dependencies if task was not resolved in time
+- resolve queue - short-lived queue to resolve task graph dependencies
+
+Interactions with those queues are handled in [`QueueService`](./src/queueservice.js).
+
+### Create task
+
+When task is created it can either be scheduled right away (no unmet dependencies) or it can stay unscheduled.
+
+```mermaid
+sequenceDiagram
+  participant client
+  participant queue as Queue API
+  participant deadlineQueue as Deadline Queue
+  participant pendingQueue as Pending Queue
+
+  client->>+queue: createTask()
+  queue-->>deadlineQueue: QueueService.putDeadlineMessage()
```

Comment:
The `alt` here makes it look like _either_ a deadline message or a pending message is enqueued, but in fact a deadline message is always enqueued. Maybe I'm just misreading the diagram?

Reply:
uhm.. right.. I was not sure how to better depict this. Will update to show that that deadline message is always added 👍